share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```
@@ -380,5 +380,58 @@ function(sysbuild_module_call)
       endforeach()
     endif()
   endforeach()
+endfunction()
+
+# Usage:
+#   sysbuild_cache_set(VAR <variable> [APPEND [LIST]] <value>)
+#
+# This function will set the specified value of the sysbuild cache variable in
+# the CMakeCache.txt file which can then be accessed by images.
+# `VAR` specifies the variable name to set/update.
+#
+# The result will be returned in `<variable>`.
+#
+# Example use:
+#   sysbuild_cache_set(VAR ATTRIBUTES APPEND LIST battery)
+#     Will add the item `battery` to the `ATTRIBUTES` variable as a new element
+#     in the list in the CMakeCache.
+#
+# <variable>: Name of variable in CMake cache.
+# APPEND:     If specified then will append the supplied data to the existing
+#             value (unless LIST is also specified, this will be treated as a
+#             string, no space character is added between appends).
+# LIST:       Will treat the append item as a list and add the new data as a
+#             new element to the list.
+# <value>:    Value to set/update.
+function(sysbuild_cache_set)
+  cmake_parse_arguments(VARS "APPEND;LIST" "VAR" "" ${ARGN})
+
+  if(NOT VARS_UNPARSED_ARGUMENTS AND VARS_APPEND)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> APPEND ...) missing required arguments to append")
+  elseif(VARS_LIST AND NOT VARS_APPEND)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> APPEND LIST ...) is required for the LIST option")
+  elseif(NOT VARS_VAR)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> ...) missing required VAR variable")
+  endif()
+
+  get_property(var_type CACHE ${VARS_VAR} PROPERTY TYPE)
+
+  # If the variable type is not set, use UNINITIALIZED which will not apply any
+  # specific formatting.
+  if(NOT var_type)
+    set(var_type "UNINITIALIZED")
+  endif()
+
+  if(VARS_APPEND)
+    set(var_new $CACHE{${VARS_VAR}})
+    if(VARS_LIST)
+      list(APPEND var_new ${VARS_UNPARSED_ARGUMENTS})
+    else()
+      set(var_new "${var_new}${VARS_UNPARSED_ARGUMENTS}")
+    endif()
+  else()
+    set(var_new ${VARS_UNPARSED_ARGUMENTS})
+  endif()
 
+  set(${VARS_VAR} ${var_new} CACHE ${var_type} "" FORCE)
```

If the variable already has a help text in the cache, then let's keep that help text by fetching the property value.
https://cmake.org/cmake/help/latest/prop_cache/HELPSTRING.html
```suggestion
  get_property(var_help CACHE ${VARS_VAR} PROPERTY HELPSTRING)
  set(${VARS_VAR} ${var_new} CACHE ${var_type} "${var_help}" FORCE)
```

---

Fixed